### ISSUE TITLE
data.aws_ec2_coip_pools.all_coip_pools.pool_ids is set of string with 63 elements

### ISSUE THREAD

**Issue by FireballDWF** [hashicorp/terraform-provider-aws#16659]
### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or other comments that do not add relevant new information or questions, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

### Terraform CLI and Terraform AWS Provider Version

Terraform v0.14.0
+ provider registry.terraform.io/hashicorp/aws v3.20.0

### Affected Resource(s)

* aws_ec2_coip_pools

### Terraform Configuration Files

Please include all Terraform configurations required to reproduce the bug. Bug reports without a functional reproduction may be closed without investigation.

```hcl
variable region {
  type = string
  default = "us-west-2"
}

provider "aws" {
  region  = var.region              
  profile = "otl05" # account with the Outpost
  alias   = "outpost"
}

data "aws_ec2_coip_pools" "all_coip_pools" {
    provider = aws.outpost
}

output coip_pools {
  value = tolist(data.aws_ec2_coip_pools.all_coip_pools.pool_ids)
}

output coip_pool_64 {
  value = tolist(data.aws_ec2_coip_pools.all_coip_pools.pool_ids)[64]
}
```

### Expected Behavior

Return all CoIP Pools that the describe-coip-pools API returns

### Actual Behavior

63 are returned, and attempt to reference 64th results in error in title

### Steps to Reproduce

1. `terraform apply`

### Important Factoids

Found while scale testing CoIP pools.  Strong suspicion that pagination not implemented by the code calling the API.  Will go collect debug to rule this theory in or out

**Comment by bflad** [comment]
Your theory is correct:

https://github.com/hashicorp/terraform-provider-aws/blob/cb4c3cf062b55a7b406f0fad862694b3cb3b7984/aws/data_source_aws_ec2_coip_pools.go#L53

Luckily the EC2 API model does already include the paginator for this API call, so the AWS Go SDK has the [`DescribeCoipPoolsPages` function](https://docs.aws.amazon.com/sdk-for-go/api/service/ec2/#EC2.DescribeCoipPoolsPages) ready for us.

**Comment by FireballDWF** [comment]
You are fast, just observed same behavior in the debug output

**Comment by FireballDWF** [comment]
As this is a blocker for me, I'm going to code up a python script to do what I need, unless you happen to produce a test version by the time I'm back at keyboard tomorrow morning, which I'm not expecting.  However, if you could produce one by 12/17, I can then test.  But after 12/18 these 512 CoIPPools is expected to be reverted back to 1.

**Comment by bflad** [comment]
Submitted: https://github.com/hashicorp/terraform-provider-aws/pull/16669 -- I'll chat with you out of band to get you a snapshot build with a few Outposts related items

**Comment by bflad** [comment]
The fix for this in the following data sources has been merged and will release with version 3.22.0 of the Terraform AWS Provider, likely tomorrow:

* `aws_ec2_coip_pools`
* `aws_ec2_local_gateways`
* `aws_ec2_local_gateway_route_tables`
* `aws_ec2_local_gateway_virtual_interface_groups`
